pkg/console/operator/status.go

```diff
@@ -157,86 +160,11 @@ func (c *consoleOperator) logConditions(conditions []operatorsv1.OperatorConditi
-		Status:  conditionStatus,
-		Reason:  conditionReason,
-		Message: conditionMessage,
-	})
-
-	return operatorConfig
-}
-
-func (c *consoleOperator) ConditionDeploymentAvailable(operatorConfig *operatorsv1.Console, message string) *operatorsv1.Console {
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:    operatorsv1.OperatorStatusTypeAvailable,
-		Status:  operatorsv1.ConditionTrue,
-		Message: message,
-	})
-
-	return operatorConfig
-}
-
-func (c *consoleOperator) ConditionDeploymentNotAvailable(operatorConfig *operatorsv1.Console, message string) *operatorsv1.Console {
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:    operatorsv1.OperatorStatusTypeAvailable,
-		Status:  operatorsv1.ConditionFalse,
-		Reason:  reasonNoPodsAvailable,
-		Message: message,
-	})
-
-	return operatorConfig
-}
-
-// ConditionResourceSyncProgressing()
-// When a sync loop aborts early:
-// - we dont know if the operand is available, so it is best not to set it.
-// - on install, an incomplete sync will mean the operator is unavailable.
-// - however, on a later sync when a change is encountered, this is not true.
-// - we do know that we encountered a component of the operand in an incorrect state
-// - we do know we are progressing because we are trying to change something about the operand
-func (c *consoleOperator) ConditionResourceSyncProgressing(operatorConfig *operatorsv1.Console, message string) *operatorsv1.Console {
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:    operatorsv1.OperatorStatusTypeProgressing,
-		Status:  operatorsv1.ConditionTrue,
-		Reason:  reasonSyncLoopProgressing,
-		Message: message,
-	})
-
-	return operatorConfig
-}
-
-func (c *consoleOperator) ConditionResourceSyncNotProgressing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:   operatorsv1.OperatorStatusTypeProgressing,
-		Status: operatorsv1.ConditionFalse,
-	})
-
-	return operatorConfig
-}
-
-// TODO: potentially eliminate the defaulting mechanism
+// TODO: eliminate the defaulting mechanism
+// This no longer has any meaning.
 func (c *consoleOperator) ConditionsDefault(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:    operatorsv1.OperatorStatusTypeAvailable,
-		Status:  operatorsv1.ConditionTrue,
-		Reason:  reasonAsExpected,
-		Message: "As expected",
-	})
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
-		Type:    operatorsv1.OperatorStatusTypeProgressing,
-		Status:  operatorsv1.ConditionFalse,
-		Reason:  reasonAsExpected,
-		Message: "As expected",
-	})
-
+	c.HandleAvailable(operatorConfig, "Default", nil)
```

Comment:
`HandleAvailable` when it receives `nil`, should set `Available:true`.  I think this will set it to false, it needs an inversion of the `Degraded` & `Progressing` logic.